x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/CIDRMatch.java

```diff
@@ -101,14 +82,81 @@ protected TypeResolution resolveType() {
     }
 
     @Override
-    public ScalarFunction makeSubstitute() {
-        ScalarFunction func = null;
+    protected Pipe makePipe() {
+        ArrayList<Pipe> arr = new ArrayList<>(addresses.size());
+        for (Expression address : addresses) {
+            arr.add(Expressions.pipe(address));
+        }
+        return new CIDRMatchFunctionPipe(source(), this, Expressions.pipe(field), arr);
+    }
+
+    @Override
+    public boolean foldable() {
+        if (field.foldable() == false) {
+            return false;
+        }
+        for (Expression address : addresses) {
+            if (address.foldable() == false) {
+                return false;
+            }
+        }
+        return true;
+    }
 
+    @Override
+    public Object fold() {
+        Object o = field.fold();
+        ArrayList<Object> arr = new ArrayList<>(addresses.size());
         for (Expression address : addresses) {
-            final Equals eq = new Equals(source(), field, address);
-            func = (func == null) ? eq : new Or(source(), func, eq);
+            arr.add(address.fold());
+        }
+        return doProcess(o, arr);
+    }
+
+    @Override
+    protected NodeInfo<? extends Expression> info() {
+        return NodeInfo.create(this, CIDRMatch::new, field, addresses);
+    }
+
+    @Override
+    public ScriptTemplate asScript() {
+        ScriptTemplate leftScript = asScript(field);
+
+        List<Object> values = new ArrayList<>(new LinkedHashSet<>(foldListOfValues(addresses, field.dataType())));
```

Comment:
```suggestion
        List<Object> values = Expressions.fold(addresses)
```

Reply:
this will not eliminate duplicated values as far as I see